### PR TITLE
Change logic

### PR DESCRIPTION
**Pull request by steffenix**

### CODE REVIEW

File: contracts/GuestList.sol [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+} from "@yearnvaults/contracts/BaseStrategy.sol";
+
+contract GuestList {
+    RegistryAPI public registry;
+    mapping(address => bool) public guests;
+    VaultAPI[] private _cachedVaults;
+    address public governance;
+
+    constructor(address _governance, address _registry) public {
+        governance = _governance;
```
[the remark this answers is not in the document]
Comment (steffenix) [reply contracts/GuestList.sol:36]: I would like to do that but it's more convenient for testing

File: contracts/GuestList.sol [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+        view
+        returns (bool)
+    {
+        if (guests[_guest] == true) {
+            return true;
+        } else if (guests[_guest] != true) {
+            return isValidStrategy(_guest);
+        }
+        return false;
+    }
```
[the remark this answers is not in the document]
Comment (steffenix) [reply contracts/GuestList.sol:51]: > ```suggestion
>     function authorized(address _guest, uint256 _amount)
>         public
>         view
>         returns (bool)
>     {
>         // Either `_guest` is pre-authorized, or it is a strategy of one of the vaults in the registry
>         return guests[_guest] || isValidStrategy(_guest);
>     }
> ```

What if guest has been blacklisted ?

File: contracts/GuestList.sol [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+
+contract GuestList {
+    RegistryAPI public registry;
+    mapping(address => bool) public guests;
+    VaultAPI[] private _cachedVaults;
+    address public governance;
+
+    constructor(address _governance, address _registry) public {
+        governance = _governance;
+        registry = RegistryAPI(_registry);
```
[the remark this answers is not in the document]
Comment (steffenix) [reply contracts/GuestList.sol:37]: That's easier for testing purposes.

File: contracts/GuestList.sol (hunk 1 above)
[the remark this answers is not in the document]
Comment (steffenix) [reply contracts/GuestList.sol:36]: understood.

File: contracts/GuestList.sol [hunk 4]
```diff
@@ -32,9 +34,9 @@ contract GuestList {
     VaultAPI[] private _cachedVaults;
     address public governance;
 
-    constructor(address _governance, address _registry) public {
-        governance = _governance;
+    constructor(address _registry) public {
         registry = RegistryAPI(_registry);
+        governance = registry.governance();
```
Comment (fubuloubu) [inline contracts/GuestList.sol:39]: Make this call each time `updateVaultCache` and `invite_guest`

**Review by fubuloubu — COMMENTED** [review]
btw don't mix `snake_case` and `camelCase`

**Pull request merged**